React peer dependency error while installing plugin-ideal-image

## 🐛 Bug Report

### Prerequisites

- [x ] I'm using the latest version of Docusaurus.
- [x ] I have tried the `npm run clear` or `yarn clear` command.
- [x ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.

### Description

I just installed docusaurus using the default classic template. Everything works well except 'plugin-ideal-image' that I tried to install. I'm getting react peer dependency error:
> Could not resolve dependency:
npm ERR! peer react@"^15.3.0 || ^16.0.0" from react-waypoint@9.0.3

It looks like the plugin is using an old version of react-waypoint (the current is 10.x that has peer dependency on react 17.x).
What are my options to install plugin-ideal-image successfully?

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## To Reproduce

If possible, use https://new.docusaurus.io to create a CodeSandbox reproducible demo of the bug.

1. Install latest version of docusaurus using the default classic template.
2. Install plugin-ideal-image
3. There should be peer dependency error reported and installation fails.

## Expected behavior

Plugin should install

## Actual Behavior

Plugin failed installating

## Your Environment

- Docusaurus version used: 2.0.0-beta.0
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0): Node 16.2.0, npm 7.13.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): Mac 11.2.3

--- a/packages/docusaurus-plugin-ideal-image/package.json
+++ b/packages/docusaurus-plugin-ideal-image/package.json
@@ -25,7 +25,7 @@
     "@docusaurus/responsive-loader": "1.4.0",
     "@docusaurus/types": "2.0.0-beta.0",
     "@endiliey/react-ideal-image": "^0.0.11",
-    "react-waypoint": "^9.0.2",
+    "react-waypoint": "^10.1.0",
     "sharp": "^0.28.2",
     "tslib": "^2.1.0",
     "webpack": "^5.28.0"

